project-base/src/Controller/Front/NewsletterController.php

```diff
@@ -53,12 +64,43 @@ public function subscribeEmailAction(Request $request)
         $form = $this->createSubscriptionForm();
         $form->handleRequest($request);
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            $email = $form->getData()['email'];
-            $this->newsletterFacade->addSubscribedEmail($email, $this->domain->getId());
+        if ($form->isSubmitted()) {
+            $this->formTimeProvider->generateFormTime($form->getName());
+
+            if ($form->isValid()) {
+                $email = $form->getData()['email'];
+
+                try {
+                    $this->newsletterFacade->addSubscribedEmail($email, $this->domain->getId());
+                    return $this->json(['success' => true]);
+                } catch (EmailWasAlreadySubscribedException $exception) {
```

Comment:
If you decide to keep the EmailAlreadySubscribed functionality, what about creating a Validator for that (similar to UniqueEmailValidator).

Because currently, when I reuse the `SubscriptionFormType` and/or `addSubscribedEmail` I can easily forget to catch this exception and application end with error 500. (plus it could make this controller method more clear)

Reply:
5a0a8da707e10b9fd774e7a0ec28d038ec6383bd